src/StringTools.cpp

```diff
@@ -0,0 +1,617 @@
+    }
+
+    void ToCamelCase(StrIter Begin, StrIter End)
+    {
+        Boole PrevCharIsWhitespace = true;
+        for( StrIter CurrIt = Begin ; CurrIt != End ; ++CurrIt )
+        {
+            Boole CurrCharIsLowerLetter = StringTools::IsLowerAlphaLetter( *CurrIt );
+            Boole CurrCharIsUpperLetter = StringTools::IsUpperAlphaLetter( *CurrIt );
+            if( PrevCharIsWhitespace && CurrCharIsLowerLetter ) {
+                (*CurrIt) = static_cast<Char8>( ::toupper(*CurrIt) );
+            }else if( !PrevCharIsWhitespace && CurrCharIsUpperLetter ) {
+                (*CurrIt) = static_cast<Char8>( ::tolower(*CurrIt) );
+            }
+            PrevCharIsWhitespace = StringTools::IsWhitespace( *CurrIt );
+        }
+    }
+
+    String CamelCaseCopy(String Source)
+    {
+        StringTools::ToCamelCase(Source.begin(),Source.end());
+        return Source;
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Size Modifying String Manipulations
+
+    void Trim(String& Source, const Boole Left, const Boole Right)
+    {
+        const String Delims = " \t\r";
+        if( Left ) {
+            size_t Forward = 0;
+            while( Forward < Source.size() )
+            {
+                if( Delims.find( Source[Forward] ) == String::npos ) {
+                    break;
+                }
+                ++Forward;
+            }
+            Source.erase(0,Forward);
+        }
+        if( Right ) {
+            size_t Reverse = Source.size();
+            while( Reverse > 0 )
+            {
+                if( Delims.find( Source[Reverse-1] ) == String::npos ) {
+                    break;
+                }
+                --Reverse;
+            }
+            Source.erase(Reverse);
+        }
+    }
+
+    String TrimCopy(String Source, const Boole Left, const Boole Right)
+    {
+        StringTools::Trim(Source,Left,Right);
+        return Source;
+    }
+
+    void RemoveDuplicateWhitespaces(String& Source)
+    {
+        static const char* Delims = " \t";
+        for( size_t CurrIndex = Source.find_first_of(Delims,0,2) ;
+             CurrIndex != String::npos ;
+             CurrIndex = Source.find_first_of(Delims,CurrIndex,2) )
+        {
+            size_t EndIndex = CurrIndex;
+            Char8 CurrWhite = Source[EndIndex];
+            while( Source[EndIndex] == CurrWhite ) EndIndex++;
+            Source.replace(CurrIndex,EndIndex - CurrIndex,1,CurrWhite);
+            CurrIndex++;
+        }
+    }
+
+    String RemoveDuplicateWhitespacesCopy(String Source)
+    {
+        StringTools::RemoveDuplicateWhitespaces(Source);
+        return Source;
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Colour Channels
+
+    Real ConvertHexToColourChannel(const StringView Hex)
+    {
+        if( Hex.size() != 2 ) {
+            String ExceptionString("Hex code requires 2 characters to express a Colour channel.");
```

Comment:
CSS has color hex code triplets. It just omits the lower precision number. If a single digit is passed we could simply duplicate that digit before resuming.